lib/AppInfo/Application.php

```diff
@@ -128,10 +128,21 @@ public function registerServices() {
 		$container->registerService('Crypt',
 			function (IAppContainer $c) {
 				$server = $c->getServer();
-				return new Crypt($server->getLogger(),
-					$server->getUserSession(),
-					$server->getConfig(),
-					$server->getL10N($c->getAppName()));
+				$hsmUrl = $this->config->getAppValue('encryption', 'hsm.url');
```

Comment:
just do document ... this causes the hsm.url app config to be two things:
1. the url for the hsm endpoint
2. the toggle for using an hsm or the internal crypto implementation

these are two different things and I have a bad feeling about it. Maybe an additional property `crypto.engine` with `internal` and `hsm` makes more sense.

Reply:
a second key would be better, yes. @sharidas please add

Reply:
Updated the PR, by setting https://github.com/owncloud/encryption/pull/90/files#diff-b8a11151e44bcffc222c5fddc9ab92c9R133

Have also verified that it works with and without hsm.